src/TextBox.js

```diff
@@ -270,7 +272,9 @@ export default class TextBox extends BaseClass {
             .attr("font-weight", d.fW)
             .style("font-weight", d.fW)
             .attr("x", `${d.tA === "middle" ? d.w / 2 : rtl ? d.tA === "start" ? d.w : 0 : d.tA === "end" ? d.w : 0}px`)
-            .attr("y", (t, i) => `${(i + 1) * d.lH - (d.lH - d.fS)}px`);
+            .attr("y", (t, i) => d.r === 0 || d.vA === "top" ? `${(i + 1) * d.lH - (d.lH - d.fS)}px` : 
+              d.vA === "middle" ? `${(d.h + d.fS) / 2 + (d.fS - d.lH)}px` : `${d.h - 2 * (d.lH - d.fS)}px`);
+
```

Comment:
This logic breaks for multi-line text, try setting `.text("My Label and a second line")` in the last example I posted.

You need to take the `i` into account, which is the current line the text is on. This should hopefully also simplify this logic.